src/classes/ApprovalProcessReminder.cls

```
@@ -62,15 +64,45 @@ global with sharing class ApprovalProcessReminder implements Database.Batchable<
                 from ProcessInstanceWorkitem 
                 where ProcessInstance.Status not in ('Approved', 'Completed')
                 and ProcessInstance.ProcessDefinition.Name != 'KBWorkflow'
-                and ActorId in :getActorIds(u.Id)
+                and ActorId in :actorIds
+                order by CreatedDate
             ];
-            if (piwis.isEmpty()) continue;
+            List<sObject> advapprs = new List<sObject>();
+            try{
+                String uId = u.Id;
+                List<sObject> advapprsIds = Database.Query('SELECT Id, sbaa__RecordField__c '
+                    + 'FROM sbaa__Approval__c '
+                    + 'WHERE sbaa__Status__c = \'Requested\' '
+                    + 'AND sbaa__Approver__c IN '
+                        + '(SELECT Id '
+                        + 'FROM sbaa__Approver__c '
+                        + 'WHERE sbaa__User__c IN :actorIds '
+                        + 'OR sbaa__GroupId__c IN :actorIds)');
+                //target record Id (which is under approval) is stored in custom fields with name stored in sbaa__RecordField__c field
+                Set<String> targetFields = new Set<String>();
+                String fieldNames = '';
+                for(sObject aa: advapprsIds){
+                    String fieldName = (String) aa.get('sbaa__RecordField__c');
+                    if(!String.isBlank(fieldName)){
+                        Boolean uniqueField = targetFields.add(fieldName);
+                        if(uniqueField){
+                            fieldNames += ', ' + fieldName;    
+                        }
+                    }
+                }
+                //query Approvals with target record Ids
+                advapprs = Database.Query('SELECT Id, CreatedDate, sbaa__RecordField__c' + fieldNames 
+                    + ' FROM sbaa__Approval__c WHERE Id IN :advapprsIds ORDER BY CreatedDate');
+            } catch (Exception ex){
+
```

probably should not ignore errors if approvals are business-critical